libnmstate/nm/applier.py

```diff
@@ -85,23 +86,73 @@ def prepare_edited_ifaces_configuration(ifaces_desired_state):
     return con_profiles
 
 
-def set_ifaces_admin_state(ifaces_desired_state):
+def set_ifaces_admin_state(ifaces_desired_state, con_profiles=()):
+    """
+    Control interface admin state by activating, deactivating and deleting
+    devices connection profiles.
+
+    The `down` and `absent` states are handled in the same manner (temporary)
+    by deactivating the device and deleting the connection profile.
+
+    For new virtual devices, the `up` state is handled by activating the
+    new connection profile. For existing devices, the device is activated,
+    leaving it to choose the correct profile.
+
+    In order to activate correctly the interfaces, the order is significant:
+    - New interfaces (virtual interfaces).
```

Comment:
Why does it need to be this order? What happens when we want to use a VLAN on top of a bond interface, doesn't the bond interface need to be activated first?

Reply:
Good point, we need new VLAN/s last in the 'new' interfaces section.
It's actually even more complex with bridges and friends.

The order matters only for masters before their slaves. Tests failed when activating the slave before the master, as the master explicit activation collided with the slaves triggering the same implicitly.
This may become a problem with the introduction of other device types and more scenarios.

Reply:
It does not need to happen in this PR, but could we instead just check the errors in case of collisions and ignore errors if it just says that we want to activate an already activated connection?